contrib/endpoints/src/api_manager/method_impl.cc

```diff
@@ -51,17 +50,19 @@ void MethodInfoImpl::addAudiencesForIssuer(const string &issuer,
   if (iss.empty()) {
     return;
   }
+
   set<string> &audiences = issuer_audiences_map_[iss];
-  stringstream ss(audiences_list);
-  string audience;
-  // Audience list is comma-delimited.
-  while (getline(ss, audience, ',')) {
-    if (!audience.empty()) {  // Only adds non-empty audience.
-      std::string aud = utils::GetUrlContent(audience);
-      if (!aud.empty()) {
-        audiences.insert(aud);
-      }
+  char *audiences_copy = new char[audiences_list.length()+1];
```

Comment:
memory leak.  
you can allocate it from stack by 

char buf[str.slze()  + 1];